File: src/commands/command.py
from abc import ABC, abstractmethod


class Command(ABC):
    """命令接口"""

    @abstractmethod
    def execute(self):
        pass

    @abstractmethod
    def undo(self):
        pass

    @abstractmethod
    def is_undoable(self) -> bool:
        """是否可撤销"""
        pass

File: src/main/java/com/editor/command/Command.java
package com.editor.command;

/**
 * 命令接口
 * 实现命令模式，支持执行、撤销和重做功能
 */
public interface Command {
    /**
     * 执行命令
     */
    void execute() throws Exception;
    
    /**
     * 撤销命令
     */
    void undo() throws Exception;
    
    /**
     * 是否可撤销
     */
    boolean isUndoable();
    
    /**
     * 获取命令描述
     */
    String getDescription();
}

